compiler/noirc_evaluator/src/ssa/function_builder/mod.rs

```diff
@@ -40,6 +41,8 @@ use super::{
 pub struct FunctionBuilder {
     pub current_function: Function,
     current_block: BasicBlockId,
+    current_block_closed: bool,
+    closed_blocked: HashSet<BasicBlockId>,
```

Comment:
What happens if we consider the block closed when we insert a terminator in general? Have we tried that approach?

Reply:
Ok, let me know which approach you think would be better - I just wanted to flag that case in case it wasn't considered.
I'd be okay with merging the original approach if you think it is better.